neural_lam/train_model.py

```diff
@@ -300,36 +308,14 @@ def main(input_args=None):
     # Only init once, on rank 0 only
     if trainer.global_rank == 0:
         utils.init_wandb_metrics(
-            logger, args.val_steps_to_log
+            logger, val_steps=args.val_steps_to_log
         )  # Do after wandb.init
-
+        # TODO: should we save the datastore config here?
+        # wandb.save()
     if args.eval:
-        if args.eval == "val":
-            eval_loader = val_loader
-        else:  # Test
-            eval_loader = torch.utils.data.DataLoader(
-                WeatherDataset(
-                    config_loader.dataset.name,
-                    pred_length=max_pred_length,
-                    split="test",
-                    subsample_step=args.step_length,
-                    subset=bool(args.subset_ds),
-                ),
-                args.batch_size,
-                shuffle=False,
-                num_workers=args.n_workers,
-            )
-
-        print(f"Running evaluation on {args.eval}")
-        trainer.test(model=model, dataloaders=eval_loader, ckpt_path=args.load)
+        trainer.test(model=model, datamodule=data_module, ckpt_path=args.load)
     else:
-        # Train model
-        trainer.fit(
-            model=model,
-            train_dataloaders=train_loader,
-            val_dataloaders=val_loader,
-            ckpt_path=args.load,
-        )
+        trainer.fit(model=model, datamodule=data_module)
```

Comment:
Why remove `ckpt_path=args.load` from this call? Does this not break resuming training from earlier checkpoint?

Reply:
I added this back in in https://github.com/leifdenby/neural-lam/commit/fc973c4c7bbef013bc2c6f19a16544a2a193a0aa, I am not sure why this was removed